src/test_report.py

```diff
@@ -20,6 +26,34 @@ def __init__(self, configs, args):
         self._db_config = configs['db_configs'][args.db_config]
         api_token = os.getenv('API_TOKEN')
         self._db = kernelci.data.get_db(self._db_config, api_token)
+        self._email_hostname = args.email_hostname
+        self._email_hostport = args.email_hostport
+
+    def create_email(self, template):
+        if template:
+            email_msg = email.mime.text.MIMEText(template, "plain", "utf-8")
+            email_msg.replace_header('content-transfer-encoding', 'quopri')
+            email_msg.set_payload(template, 'utf-8')
+            email_msg['To'] = os.getenv('EMAIL_TO')
+            email_msg['From'] = os.getenv('EMAIL_FROM')
+            email_msg['Subject'] = os.getenv('EMAIL_SUBJECT')
+            self._print(email_msg)
+        return email_msg
+
+    def send_email(self, email_msg):
+        try:
+            server_user = os.getenv('SERVER_USER')
+            server_password = os.getenv('SERVER_PASSWORD')
+            email_server = smtplib.SMTP(self._email_hostname,
+                                        self._email_hostport)
+            email_server.starttls()
+            email_server.login(server_user, server_password)
+            email_server.send_message(email_msg)
+            self._print("Email sent.")
+        except smtplib.SMTPException as e:
+            self._print("Check email credentials.")
+        else:
+            email_server.quit()
```

Comment:
@mgalka I am not entirely sure about if we can keep `quit()` function in a else, maybe could be interesting handle exceptions and then use `finally`.

Reply:
updated